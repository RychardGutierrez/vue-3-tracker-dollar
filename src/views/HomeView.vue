<template>
  <main class="container text-black">

    <SearchBar>
      <template v-slot="{ countrySearchResults, searchError }">
        <ListResult :countrySearchResults="countrySearchResults" :searchError="searchError"
          :goToCountry="goToCountry" />
      </template>
    </SearchBar>

    <div class="flex flex-col gap-4">
      <Suspense>
        <CountryList />
        <template #fallback>
          <p>Loading...</p>
        </template>
      </Suspense>
    </div>
  </main>
</template>

<script setup>
import CountryList from '../components/country/CountryList.vue';
import SearchBar from '../components/ui/SearchBar.vue';
import ListResult from '../components/ui/ListResult.vue';

import useRouteCountry from '../composables/useRouteCountry';

const { goToCountry } = useRouteCountry();

</script>
