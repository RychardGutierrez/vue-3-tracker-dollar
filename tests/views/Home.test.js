import { shallowMount } from '@vue/test-utils';
import HomeView from '../../src/views/HomeView.vue';

jest.mock('../../src/composables/useRouteCountry', () => {
  const useRouteCountry = () => ({
    goToCountry: jest.fn(),
  });
  return useRouteCountry;
});

jest.mock('../../src/components/country/CountryList.vue', () => jest.fn());

describe('HomeView', () => {
  test('should to match with snapshot', () => {
    const wraper = shallowMount(HomeView);

    expect(wraper.html()).toMatchSnapshot();
  });
});
