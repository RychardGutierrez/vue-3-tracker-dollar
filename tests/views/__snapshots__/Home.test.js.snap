// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeView should to match with snapshot 1`] = `
"<main class=\\"container text-black\\">
  <search-bar-stub></search-bar-stub>
  <div class=\\"flex flex-col gap-4\\">
    <suspense-stub></suspense-stub>
  </div>
</main>"
`;
